ui/src/components/canvas/FlowCanvas.tsx

```diff
@@ -87,12 +86,28 @@ const FlowCanvas = () => {
     updateLayoutDensity,
   } = useAppActions()
 
-  useKeyboardShortcuts()
-
   useEffect(() => {
     reactFlowInstance.fitView()
   }, [layout, reactFlowInstance])
 
+  const onUndoRedoKeyDown = (event: KeyboardEvent) => {
```

Comment:
Since this function is no longer a hook, we can avoid re-defining it on every render by moving it out of the `FlowCanvas` component and passing in the undo and redo functions as arguments.